datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -618,6 +621,7 @@ protected void commitWithRetries(Consumer<?, ?> consumer, Optional<Map<TopicPart
         }
         _logger.info("Commit succeeded.");
       } catch (KafkaException e) {
+        lastKafkaExceptionRef.set(e);
         if (_shutdown) {
           _logger.info("Caught KafkaException in commitWithRetries while shutting down, so exiting.", e);
           return true;
```

Comment:
During shutdown, if it hits KafkaException, it returns true, which does not really mean success. This can impact the postCommit hook.

Reply:
This is no longer relevant now that post-commit hook is removed.